packages/jest-haste-map/src/crawlers/watchman.ts

```diff
@@ -148,6 +158,8 @@ export = async function watchmanCrawl(
     // files.
     if (watchmanFileResults.isFresh) {
       files = new Map();
+      deprecatedFiles = new Map(data.files);
```

Comment:
If watchman crashes and gives us a fresh instance, we just start from scratch, right? If so, this seems like unnecessary bookkeeping. Could you explain the idea behind this?

Reply:
If Watchman crashes, I'm just getting a full list of files. I need to also know what files were deleted because I have to pass them back for use here:
https://github.com/facebook/jest/blob/master/packages/jest-haste-map/src/index.ts#L642

Reply:
In this code path (when Watchman is fresh), that map has all existing files removed as we iterate through the files, so it ends up being just a list of removed files and nothing else.